core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -184,6 +184,27 @@ void *impl_allocate_common(const int device_id,
   cudaError_t error_code = cudaSuccess;
 #ifndef CUDART_VERSION
 #error CUDART_VERSION undefined!
+#elif defined(KOKKOS_ENABLE_IMPL_CUDA_EMULATE_UNIFIED_MEMORY)
+  // This is inteded to simulate Grace-Hopper like behavior
+  error_code = cudaMallocManaged(&ptr, arg_alloc_size, cudaMemAttachGlobal);
+  KOKKOS_IMPL_CUDA_SAFE_CALL(cudaDeviceSynchronize());
```

Comment:
Please confirm that the `cudaDeviceSynchronize()` would not return an error if `error_code` was not `cudaSuccess`.